akka-actor/src/main/scala/akka/pattern/BackoffOptions.scala

```diff
@@ -125,14 +183,136 @@ object Backoff {
    * @param randomFactor after calculation of the exponential back-off an additional
    *   random delay based on this factor is added, e.g. `0.2` adds up to `20%` delay.
    *   In order to skip this additional delay pass in `0`.
+   * @param maxRestartAttempts maximum number of attempts to restart the child actor.
+   *   In order to restart infinitely pass in `-1`
    */
+  def onFailure(
```

Comment:
Might be nice to mark this 'Java API' in the scaladoc while we're here

Reply:
Done